### PR TITLE
Feature: create entity, read entity by name and create entity alias

### PR DESCRIPTION
**Pull request by slckl**
Probably replaces #38 

Have the same need, entity and entity-alias allows applying vault policies to an identity.

Cleaned up the previous work by adding docs, and trimming some unnecessary changes.
I've also implemented an extra api call for obtaining entity by name, which is included in the test from the previous PR.

Could probably implement more of the identity API (somewhat easy thanks to the design of this library, good job), but have no use case for the rest of it right now.

### CODE REVIEW

File: src/api/entity_alias/responses.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+use serde::{Deserialize, Serialize};
+
+/// Response from executing
+/// [CreateEntityAliasRequest](crate::api::entity_alias::request::CreateEntityAliasRequest)
+#[derive(Deserialize, Debug, Serialize)]
+pub struct CreateEntityAliasResponse {
+    pub request_id: String,
+    pub lease_id: String,
+    pub renewable: bool,
+    pub lease_duration: i64,
```
Comment (Haennetz) [inline src/api/entity_alias/responses.rs:10]:
```suggestion
    pub lease_duration: u64,
```

**Review by Haennetz — COMMENTED** [review]
Hey Thanks for your pull request, and sorry for the delay.
We should add a module `identity` and add submodules for `entity` and `entity_alias`.
It would be nice to provide a feature complete implementation for the `entity` and `entity_alias` endpoints.
If you have the time it would be great if you can add the identity API completely, otherwise I will create a new issue to do that.

**Comment by slckl** [comment]
Hey, glad to see some activity here.
I'm still using my fork for this in production, so I would love to upstream this.

Will check back next week how much I can do wrt the whole `identity` endpoint.

**Comment by Haennetz** [comment]
It would be enough  to add at least the entity and entity_alias endpoints. The rest of the identity secret engine kann be done later on.

**Comment by stormshield-gt** [comment]
I would be glad to give you a hand with this if you need any help. I'm also needing this right now, and the group endpoint too. I could advance this in my own branch if that's OK.

**Comment by stormshield-gt** [comment]
@slckl I hope you don't mind, I didn't wait for your response and continue the work in another PR. I supposed you will be fine to have somebody continue the work. I've credited you on the first commits

**Comment by slckl** [comment]
@stormshield-gt sounds good. It sounds like you're more eager for this than I am at the moment, so, ofc, go for it.
We can close this one, when your branch is merged probably.

**Comment by Haennetz** [comment]
I close this issue as well because we merged #77

**Pull request closed without merging**